### ISSUE TITLE
pfscan error not catched

### ISSUE THREAD

**Issue by dvallenet** [ebi-pf-team/interproscan#97]
We have random segfault errors with pfscan on centos6 that are not catched by ps_scan.pl as it uses open2 to run pfscan with a pipe without checking the exit status (see code below).
Thus iprscan does not raise any error and we get no PROSITE result.

ps_scan.pl code :
```
        require IPC::Open2;
        my ($reader, $writer);
        my $cmd = "@pre_command - @post_command";
        my $pid =
        eval {
            IPC::Open2::open2($reader, $writer, $cmd)
            or die "Could not fork pipe to $cmd: $!";
        };
        if ($@) {
            die "$@\n" . '>' x 62 . "\nERROR: 'pfscan' execution failed.
            Check pfscan is in your PATH\n" . '>' x 62 . "\n";
        }
        local $/ = \32767;# buffer size
        print $writer ">seq for pfscan\n";
        print $writer "$1\n" while $sequence =~ /(.{1,60})/g;
        close $writer;
        $out = scanProfiles($reader, $level_min-1);
        close $reader or die "Error $? with $cmd";
        waitpid $pid, 0; #avoid defunct kid processes 
        #exit status should be checked here see https://perldoc.perl.org/IPC/Open2.html but I am not sure that it will handle all the errors correctly
```
As a fix, we  force the use of system command instead of open2 by setting $NO_DIRECT_PIPE=1 in ps_scan.pl :
```
eval { require IPC::Open2 };
my $NO_DIRECT_PIPE=$? || $^O eq "MSWin32" ? 1 : 0;
$NO_DIRECT_PIPE=1; #force not use of Open2 because exit status of pfscan is not checked in this mode
```

**Comment by gsn7** [comment]
thanks for reporting this. we will forward this bug to the Prosite team

**Comment by flass** [comment]
Hi,

I just got a related error.

Using the following command:
```sh
$ interproscan -T tmp/interpro -i all_proteomes.nr.faa.0 -b all_proteomes.nr.faa.0 --formats TSV --cpu 15 --iprlookup --goterms --pathways
```
I got the following log:
```
09/10/2019 09:12:05:355 Welcome to InterProScan-5.38-76.0
09/10/2019 09:12:05:356 Running InterProScan v5 in STANDALONE mode... on Linux
09/10/2019 09:12:19:536 Loading file all_proteomes.nr.faa.0
09/10/2019 09:12:19:538 Running the following analyses:
[CDD-3.17,Coils-2.2.1,Gene3D-4.2.0,Hamap-2019_01,MobiDBLite-2.0,Pfam-32.0,PIRSF-3.02,PRINTS-42.0,ProSitePatterns-2019_01,ProSiteProfiles-2019_01,SFLD-4,SMART-7.1,SUPERFAMILY-1.75,TIGRFAM-15.0]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
09/10/2019 09:13:16:209 28% completed
09/10/2019 09:13:22:247 57% completed
09/10/2019 09:13:26:550 85% completed
ERROR: uk.ac.ebi.interpro.scan.management.model.implementations.RunBinaryStep - Command line failed with exit code: 129
Command: bin/prosite/ps_scan.pl -d data/prosite/2019_01/prosite.dat -w bin/prosite/pfsearch -s -m -o gff -y data/prosite/2019_01/prosite_models tmp/interpro/annotvibrio16cpu_20191009_091209789_9vyg//jobPrositeProfiles/000000009001_000000010000.fasta
Error output from binary:

```
As mentioned in the bug report above, there is no error (or rather an empty error message) reported from the `ps_scan.pl` binary.

Please find attached the temporary files of the PrositeProfile search:
[failed_jobPrositeProfiles.tar.gz](https://github.com/ebi-pf-team/interproscan/files/3706989/failed_jobPrositeProfiles.tar.gz)

Thanks in advance for the help.

**Comment by dvallenet** [comment]
I don't think your error is related to the problem I mentioned

I found the origin of the pfscan segmentation fault. 
This happens when the path of the input files exceeds a certain number of characters (about more than 250 characters if I remember).  It looks like a hard-coded constant in the fortran code.
As a solution, we now execute interproscan in /tmp or /dev/shm on cluster nodes.

**Comment by duboism** [comment]
`ps_scan.pl` has been updated (see sib-swiss/pftools3#9) to detect such errors. This version has not yet been released however.

Concerning the problem in `pfscan` itself, the SIB team recommends to use `pftoolsV3` (see sib-swiss/pftools3#8). Is there a way to do that in Interpro ?

**Comment by gsn7** [comment]
@duboism thanks for this update. We will test the new version

**Comment by duboism** [comment]
Good news. Note that in the meantime, the pftools team at SIB released version v3.2.5 that includes the updated `ps_scan.pl`.

**Comment by duboism** [comment]
Any progress on this issue ?

**Comment by duboism** [comment]
Good news. Is it in version `5.51-85.0` ? If no, do you know when this will be released ?

**Comment by duboism** [comment]
Answering my own question: yes, is in `5.51-85.0` (at least).

**Comment by duboism** [comment]
And more precisely, it was first released in `5.48-83.0`.
